tests/components/prusalink/test_init.py

```diff
@@ -50,3 +58,51 @@ async def test_failed_update(
 
     for state in hass.states.async_all():
         assert state.state == "unavailable"
+
+
+async def test_migration_1_2(hass: HomeAssistant, mock_api) -> None:
+    """Test migrating from version 1 to 2."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_HOST: "http://prusaxl.local",
+            CONF_API_KEY: "api-key",
+        },
+        version=1,
+    )
+    entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    config_entries = hass.config_entries.async_entries(DOMAIN)
+
+    # Ensure that we have username, password after migration
+    assert len(config_entries) == 1
+    assert config_entries[0].data == {
+        CONF_HOST: "http://prusaxl.local",
+        CONF_USERNAME: "maker",
+        CONF_PASSWORD: "api-key",
+    }
+
+
+async def test_outdated_firmware_migration_1_2(hass: HomeAssistant, mock_api) -> None:
+    """Test migrating from version 1 to 2."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_HOST: "http://prusaxl.local",
+            CONF_API_KEY: "api-key",
+        },
+        version=1,
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "pyprusalink.PrusaLink.get_info",
+        side_effect=InvalidAuth,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert entry.state == ConfigEntryState.SETUP_ERROR
```

Comment:
Please assert also that the repair issue is created

Reply:
Added. Additionally, added a test case to verify that the issue is also removed on reload with working api. https://github.com/home-assistant/core/pull/103396/commits/b482c2b448e6e7427f06843410b335e1563a928c

Thanks!